src/main/scala/strawman/collection/View.scala

```diff
@@ -250,6 +255,19 @@ object View extends IterableFactory[View] {
 
 }
 
+object ViewTransformer {
+  type VT[-A,+B] = Iterable[A] => View[B]
+  implicit class ViewTransFormerOps[A,B](val vt: VT[A,B]) {
+    def map[C](f: B => C): VT[A,C] = v => new View.Map(vt(v),f)
+    def flatMap[C](f: B => IterableOnce[C]): VT[A,C] = v => new View.FlatMap(vt(v),f)
+    def filter(f: B => Boolean): VT[A,B] = v => new View.Filter(vt(v),f)
+    def drop(n: Int): VT[A,B] = v => new View.Drop(vt(v),n)
+    def collect[C](pf: scala.PartialFunction[B,C]): VT[A,C] = v => new View.Collect(vt(v),pf)
+    def zipwithIndex: VT[A,(B,Int)] = v => new View.ZipWithIndex[B](vt(v))
+  }
```

Comment:
What’s clunkier exactly? You can also define a factory method in the `ViewTransformer` companion:

~~~ scala
object ViewTransformer {
  def apply[A, B](f: Iterable[A] => View[B]): ViewTransformer[A, B] = …
}

class ViewTransformer[-A, +B] {
  def apply(as: Iterable[A]): View[B]
  def map[C](f: B => C) = ViewTransformer(it => View.Map(apply(v)), f)
}
~~~

Or, you can also use the “lambda” syntax (which has been generalized to SAM types):

~~~ scala
trait ViewTransformer[-A, +B] {
  def apply(as: Iterable[A]): View[B]
  def map[C](f: B => C): ViewTransformer[A, C] = as => View.Map(apply(as), f)
}
~~~